python/semantic_kernel/processes/process_types.py

```diff
@@ -0,0 +1,34 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+from typing import Any, TypeVar, Union, get_args, get_origin, get_type_hints
+
+from semantic_kernel.processes.kernel_process.kernel_process_step import KernelProcessStep
+
+TStep = TypeVar("TStep", bound=KernelProcessStep)
+TState = TypeVar("TState")
+
+
+def get_generic_state_type(cls) -> Any:
```

Comment:
interesting to see that neither python nor .NET gave us a break on extracting the generic types 😅